lua/orgmode/objects/calendar.lua

```diff
@@ -184,14 +201,34 @@ function Calendar:render()
     table.insert(content, ' [i] - enter date')
   end
 
+  -- FIXME this line is currently not shown because of a bug.
+  if self:has_time() then
+    table.insert(content, ' [t] - enter time  [T] - clear time')
+  else
+    table.insert(content, ' [t] - enter time')
+  end
+
   vim.api.nvim_buf_set_lines(self.buf, 0, -1, true, content)
   vim.api.nvim_buf_clear_namespace(self.buf, namespace, 0, -1)
   if self.clearable then
     vim.api.nvim_buf_add_highlight(self.buf, namespace, 'Comment', #content - 3, 0, -1)
   end
+
+  if not self:has_time() then
+    vim.api.nvim_buf_add_highlight(self.buf, namespace, 'Comment', 8, 0, -1)
+  end
+
+  vim.api.nvim_buf_add_highlight(self.buf, namespace, 'Comment', #content - 4, 0, -1)
+  vim.api.nvim_buf_add_highlight(self.buf, namespace, 'Comment', #content - 3, 0, -1)
   vim.api.nvim_buf_add_highlight(self.buf, namespace, 'Comment', #content - 2, 0, -1)
   vim.api.nvim_buf_add_highlight(self.buf, namespace, 'Comment', #content - 1, 0, -1)
 
+  -- highlight the cell of the current day
+  self:highlight_day(content, Date.today(), 'OrgCalendarToday')
+  -- highlight selected day
+  self:highlight_day(content, self.date, 'OrgCalendarSelected')
+
+  -- TODO this new loop (after Kristjans refactoring) is currently not fully understood.
```

Comment:
@kristijanhusak I am struggeling a bit to fully understand, what your new version of the rendering adjustment does differently than before and how it behaves differently to my `highlight_day` method. I would appreciate your help to consolidate your loop and `on_render_day` method with my `highlight_day` method.

Reply:
@seflue it's somewhat similar. You can just put your logic inside the `on_render_day` in a same way the today's date is highlighted. For each day `on_render_day` is triggered with a date and some positioning info. That should allow you to highlight any of the days however you want.

Reply:
Understood. I integrated my code with yours.